apps/anoma_node/lib/node/transaction/backends.ex

```diff
@@ -381,4 +392,89 @@ defmodule Anoma.Node.Transaction.Backends do
 
     EventBroker.event(event)
   end
+
+  @spec cairo_resource_tx(String.t(), binary(), Noun.t()) ::
+          :ok | :error
+  defp cairo_resource_tx(node_id, id, result) do
+    with {:ok, tx} <- CTransaction.from_noun(result),
+         true <- Anoma.RM.Transaction.verify(tx),
+         true <- root_existence_check(tx, node_id, id),
+         # No need to check the commitment existence
+         true <- nullifier_existence_check(tx, node_id, id) do
+      {ct, append_roots} =
+        case Ordering.read(node_id, {id, :ct}) do
+          :absent ->
+            {CTransaction.cm_tree(),
+             MapSet.new([Anoma.Constants.default_cairo_rm_root()])}
+
+          val ->
+            {val, MapSet.new()}
+        end
+
+      {ct_new, anchor} =
+        CommitmentTree.add(ct, tx.commitments)
+
+      Ordering.add(
+        node_id,
+        {id,
+         %{
+           append: [
+             {:nullifiers, MapSet.new(tx.nullifiers)},
+             {:commitments, MapSet.new(tx.commitments)},
+             {:roots, MapSet.put(append_roots, anchor)}
+           ],
+           write: [{:anchor, anchor}, {:ct, ct_new}]
```

Comment:
```suggestion
           write: [{:cairo_anchor, anchor}, {:cairo_ct, ct_new}]
```

Reply:
also: why are you storing an achor here? we store it in the TRM to check that commitments were included at a specific timestamp. you seem to be storing roots separately and not using the anchor directly anywhere

Reply:
removed the anchor